addons/udes_stock/models/common.py

```diff
@@ -0,0 +1,46 @@
+import re
+
+
+def get_next_name(obj, code, sequence=None):
+    """
+    Get the next name for an object.
+    For when we want to create an object whose name links back to a previous
+    object.  For example BATCH/00001-002.
+    Assumes original names are of the form `r".*\d+"`.
+    Arguments:
+        obj - the source object for the name
+        code - the code for the object's model in the ir_sequence table
+        sequence - the relevant ir sequence to use
+    Returns:
+        The generated name, a string.
+    """
+    IrSequence = obj.env["ir.sequence"]
+
+    if not sequence:
+        # Use the IrSequence code in 14 approach.
+        # Return False if no sequence found to be consistent with Odoo Core.
+        obj.check_access_rights("read")
+        force_company = obj._context.get("force_company")
+        if not force_company:
+            force_company = obj.env.user.company_id.id
+        
+        seq_ids = IrSequence.search([('code', '=', code), ('company_id', 'in', [force_company, False])], order='company_id')
+        if not seq_ids:
+            return False
+        ir_sequence = seq_ids[0]
+    else:
+        ir_sequence = sequence
+    
+    # Name pattern for continuation object.
+    # Is three digits enough? Model configurable?
+    name_pattern = r"({}\d+)-(\d{{3}})".format(re.escape(ir_sequence.prefix))
```

Comment:
For regex pernickityness, I'd use `[0-9]` for the dgits, or set `re.ASCII` in the call to `re.match`.

Reply:
There's two \ds :smiley: